src/digest.c

```diff
@@ -116,66 +178,52 @@ SEXP digest(SEXP Txt, SEXP Algo, SEXP Length, SEXP Skip, SEXP Leave_raw, SEXP Se
     switch (algo) {
     case 1: {     /* md5 case */
         md5_context ctx;
-        output_length = 16;
-        unsigned char md5sum[16];
-        int j;
+        output_length = 16; // produces 16*8 = 128 bits
+        unsigned char md5sum[output_length];
```

Comment:
I really appreciate this and similar cleanups.

Reply:
You're welcome. I *think* these kind of normalization changes are going to point the way to larger scale DRY opportunities, which should then make the other "consider X" issues much easier.